src/panels/lovelace/common/long-press.ts

```diff
@@ -0,0 +1,98 @@
+import {render, directive} from 'lit-html';
+
+const isTouch =
+  "ontouchstart" in window ||
+  navigator.MaxTouchPoints > 0 ||
+  navigator.msMaxTouchPoints > 0;
+
+const longPressSetup = () => {
+  if(document.body.longPress) return document.body.longPress;
+
+  const longPress = document.createElement("div");
+  document.body.appendChild(longPress);
+  document.body.longPress = longPress;
+  Object.assign(longPress.style, {
+    position: "fixed",
+    width: isTouch ? "100px" : "50px",
+    height: isTouch ? "100px" : "50px",
+    transform: "translate(-50%, -50%)",
+    pointerEvents: "none",
+  });
+  longPress.ripple = document.createElement("mwc-ripple");
+  longPress.appendChild(longPress.ripple);
+  longPress.ripple.primary = true;
+
+  longPress.timer = null;
+  longPress.held = false;
+
+  longPress.start = (x,y) => {
+    Object.assign(longPress.style, {
+      top: `${y}px`,
+      left: `${x}px`,
+      visibility: 'visible',
+    });
+    longPress.ripple.disabled = false;
+    longPress.ripple.active = true;
+    longPress.ripple.unbounded = true;
+  };
+
+  longPress.stop = () => {
+    longPress.ripple.active = false;
+    longPress.ripple.disabled = true;
+    longPress.style.visibility = "hidden";
+  };
+
+  [
+    isTouch ? "touchcancel" : "mouseout",
+    "mouseup",
+    "touchmove",
+    "mousewheel",
+    "wheel",
+    "scroll",
+  ].forEach((ev) => {
+    document.addEventListener(ev, () => {
+      clearTimeout(longPress.timer);
+      longPress.stop();
+    });
+  });
+
+  return longPress;
+};
+
+export const longPressBind = (element, onClick=null, onHold=null) => {
+  const longPress = longPressSetup();
+  if(!longPress) return;
+
+  const firstSetup = (element.longPress == null);
+
+  element.longPress = {
+    onClick: onClick,
+    onHold: onHold,
+  };
+
+  if(!firstSetup) return;
+
+  element.addEventListener(isTouch ? "touchstart" : "mousedown",
```

Comment:
make sure to add these click listeners as passive, since we're not going to call preventDefault. That way the browser can optimize. https://developer.mozilla.org/en-US/docs/Web/API/EventTarget/addEventListener#Parameters